Beeper/BarcelonaMautrixIPC/Runtime/UnixSocketMautrixIPCChannel.swift

```diff
@@ -0,0 +1,83 @@
+import Foundation
+import Socket
+import Swog
+
+fileprivate let log = Logger(category: "UnixSocketMautrixIPCChannel")
+
+public class UnixSocketMautrixIPCChannel: MautrixIPCInputChannel, MautrixIPCOutputChannel {
+    private let socket: Socket
+    private var readThread: Thread? = nil
+    private var readRunLoop: CFRunLoop? = nil
+    private var readCallback: ((Data) -> ())? = nil
+    
+    public init(_ socketPath: String) {
+        do {
+            try socket = Socket.create(family:.unix)
+            try socket.connect(to: socketPath)
+            log.info("Connected to unix socket \(socketPath)")
+        } catch let error {
+            fatalError("Failed to connect unix socket \(error)")
+        }
+    }
+    
+    private func setupReadThread() {
+        readThread = Thread {
+            self.readRunLoop = CFRunLoopGetCurrent()
+            
+            RunLoop.current.schedule {
+                do {
+                    var shouldKeepRunning = true
+                    
+                    repeat {
+                        log.info("Waiting for data")
+                        var readData = Data(capacity: 4096)
+                        let bytesRead = try self.socket.read(into: &readData)
+                        log.info("Read \(bytesRead) bytes from the unix socket")
+                        
+                        if bytesRead > 0 {
+                            if let readCallback = self.readCallback {
+                                readCallback(readData)
+                            }
+                        }
+                        
+                        if bytesRead == 0 {
```

Comment:
Would this make more sense as an `else` to the `if bytesRead > 0`?

Reply:
Or a `guard`?